### DISCUSSION TITLE
Better way to call both actions and state through useShallow

### DISCUSSION THREAD

**Discussion by ImamJanjua** [pmndrs/zustand#2543] [General]
Hi, at the moment i am getting my states and actions from from useShallow hook at same way to get all on one swipe, but is there a better way? 

`const [
    createStartSlot,
    setCreateStartSlot,
    createSlotLength,
    setCreateSlotLength,
    createEndSlot,
    setCreateEndSlot
  ] = useCalendarStore(
    useShallow((state) => [
      state.createStartSlot,
      state.setCreateStartSlot,
      state.createSlotLength,
      state.setCreateSlotLength,
      state.createEndSlot,
      state.setCreateEndSlot,
    ]),
  );`

**Comment by dai-shi** [comment]
I would prefer something like this, but anything would be fine:
```js
const useFooState = create((set) => ({
  foo: 'foo',
  bar: 'bar',
  setFoo: () => { /* ... */ },
  setBoo: () => { /* ... */ },
}));

const Component = () => {
  const foo = useFooStore((state) => state.foo);
  const bar = useFooStore((state) => state.bar);
  const { setFoo, setBar } = useFooStore.getState();
  // ...
}
```
